test/p3_tests.jl

```diff
@@ -1,42 +1,61 @@
-import Test as TT
-import ClimaParams
-import CloudMicrophysics as CM
+using Test: @testset, @test, @test_throws
 import CloudMicrophysics.P3Scheme as P3
 import CloudMicrophysics.Parameters as CMP
 import CloudMicrophysics.Microphysics2M as CM2
 import Thermodynamics as TD
-
-import QuadGK as QGK
+import ClimaParams as CP
 
 @info("P3 Scheme Tests")
 
-function test_incomplete_gamma_function_approximation(FT)
-    # A smoke tests to see if the reference values are not changed
-    TT.@testset "Incomplete Γ function approximation" begin
-
-        Γ_lower_reference = [
-            [0.6171173028812539, 0.9999647375921944, 0.9999999999999561],
-            [66.43238525128415, 192195.93434392574, 362553.66f0],
-            [9.801186821098075e11, 8.309791f14, 1.1845235f17],
-        ]
-
-        a = [FT(1), FT(10), FT(20)]
-        z = [FT(1), FT(10), FT(30)]
-
-        for it in [1, 2, 3]
-            for jt in [1, 2, 3]
-                TT.@test P3.Γ_lower(a[it], z[jt]) ≈ Γ_lower_reference[it][jt] rtol =
-                    1e-3
-            end
-        end
+function test_p3_state_creation(FT)
+    @testset "P3State Creation and Properties" begin
+        # Test creating a state with valid parameters
+        FT = Float64
+        params = CMP.ParametersP3(FT)
+
+        # Test unrimed state
+        state_unrimed = P3.get_state(params; F_rim = FT(0.0), ρ_r = FT(400))
+        @test P3.isunrimed(state_unrimed)
+        @test !P3.isrimed(state_unrimed)
+
+        # Test rimed state
+        state_rimed = P3.get_state(params; F_rim = FT(0.5), ρ_r = FT(400))
+        @test P3.isrimed(state_rimed)
+        @test !P3.isunrimed(state_rimed)
+
+        # Test thresholds for unrimed state
+        @test length(P3.threshold_tuple(state_unrimed)) == 1
+        @test P3.threshold_tuple(state_unrimed)[1] == state_unrimed.D_th
+
+        # Test thresholds for rimed state
+        @test length(P3.threshold_tuple(state_rimed)) == 3
+        @test P3.threshold_tuple(state_rimed) ==
+              (state_rimed.D_th, state_rimed.D_gr, state_rimed.D_cr)
+
+        # Test parameter boundary validation
+        @test_throws AssertionError P3.get_state(
+            params;
+            F_rim = FT(-0.1),
+            ρ_r = FT(400),
+        )
+        @test_throws AssertionError P3.get_state(
+            params;
+            F_rim = FT(1),
+            ρ_r = FT(400),
+        )
+        @test_throws AssertionError P3.get_state(
+            params;
+            F_rim = FT(0.5),
+            ρ_r = FT(-400),
+        )
     end
 end
 
 function test_thresholds_solver(FT)
 
-    p3 = CMP.ParametersP3(FT)
+    params = CMP.ParametersP3(FT)
```

Comment:
Thinking about coupling with Atmos model. `params` is usually a name used for the struct with all parameters. We might want to use a different name here, maybe?